docs/develop/databases.md

```diff
@@ -85,3 +83,8 @@ In the cloud, how the database is provisioned depends on the type of [Encore Env
 Encore automatically provisions databases to match what your application requires.
 Simply define a database schema ([see above](#defining-a-database-schema)) and Encore
 will provision the database at the start of the next deploy.
+
+
+## Handling dirty database migrations
+
+When a database migration doesn't apply cleanly, `golang-migrate` (the library we use under the hood to manage database migrations) marks the migration state as "dirty" and refuses to continue. This can be fixed by editing the database table (fixing the schema problem and then running `UPDATE schema_migrations SET dirty = false`).
```

Comment:
I think this section needs to be expanded on to provide more of a thorough explanation of what dirty migrations are, why they exist and so on. We don't need to mention `golang-migrate` since it's an implementation detail, but we should cover the semantics in greater detail.

Reply:
Okay, i will do that ASAP